addons/account/models/chart_template.py

```diff
@@ -176,7 +176,15 @@ def try_loading_for_current_company(self):
             for template in self:
                 template.load_for_current_company(15.0, 15.0)
 
-    def load_for_current_company(self, sale_tax_rate, purchase_tax_rate):
+    def try_loading_for_company(self, company):
+        if not company:
+            self.try_loading_for_current_company()
+        else:
+            if not company.chart_template_id and not self.existing_accounting(company):
+                for template in self:
+                    template.with_context(default_company_id=company.id).load_for_current_company(15.0, 15.0, company)
+
+    def load_for_current_company(self, sale_tax_rate, purchase_tax_rate, company=False):
```

Comment:
I find it strange and confusing that the method is called `load_for_current_company` but can now be called for another company than the current one if you pass a company parameter.
So since we are refactoring why not only keep the 2 following methods: `load_for_company` and `try_loading_for_company` but it means that you'll have to change the code of all chart_templates that were previously calling `try_loading_for_current_company` and `load_for_current_company`. Also the new methods that are taking a company as parameter should works as follow: if no company is provided, assume that we are calling with the current company.
This would be the best solution as we would simplify the code and won't have duplicated method but if you don't want to change the code on each localization, I would be fine keeping the `try_loading_for_current_company` and `load_for_current_company` but those function would just call their counterpart `try_loading_for_company` and `load_for_company` without passing the company parameter.

TLDR; the logic should be inside `try_loading_for_company` and `load_for_company`